library/src/main/java/com/novoda/downloadmanager/lib/ContentLengthFetcher.java

```diff
@@ -0,0 +1,77 @@
+package com.novoda.downloadmanager.lib;
+
+import android.util.Pair;
+
+import com.novoda.notils.logger.simple.Log;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.concurrent.TimeUnit;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+
+public class ContentLengthFetcher {
+
+    private static final String HEADER_CONTENT_LENGTH = "Content-Length";
+    private static final int UNKNOWN_CONTENT_LENGTH = -1;
+    private static final int TIMEOUT_MILLIS = (int) TimeUnit.SECONDS.toMillis(10);
+    private static final String HEADER_USER_AGENT = "User-Agent";
+    private static final String METHOD_HEAD = "HEAD";
+
+    public long fetchContentLengthFor(DownloadInfo info) {
+        HttpURLConnection conn = null;
+        try {
+            conn = (HttpURLConnection) new URL(info.mUri).openConnection();
+            conn.setInstanceFollowRedirects(true);
+            conn.setConnectTimeout(TIMEOUT_MILLIS);
+            conn.setReadTimeout(TIMEOUT_MILLIS);
+            conn.setRequestMethod(METHOD_HEAD);
+            addRequestHeaders(info, conn);
```

Comment:
Following the [same pattern](https://github.com/novoda/download-manager/blob/master/library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java#L308) when downloading, we add the request headers for the head request as well.